cirq-core/cirq/protocols/json_serialization.py

```diff
@@ -505,6 +536,22 @@ def get_serializable_by_keys(obj: Any) -> List[SerializableByKey]:
     return []
 
 
+def json_cirq_type(obj: Any) -> str:
+    """Returns the string type of `obj` used in JSON serialization.
+
+    Types can provide custom string types with `_json_cirq_type_`; otherwise, a
+    Cirq type will use its type name as its string type.
+
+    Non-Cirq types do not have a json_cirq_type value.
+    """
+    if hasattr(obj, '_json_cirq_type_'):
+        return obj._json_cirq_type_()
+    obj_type = obj if isinstance(obj, type) else type(obj)
+    if obj_type.__module__[:4] == 'cirq':
```

Comment:
nit: maybe just a bit cleaner to use python's `startswith` method?

Reply:
Fixed.